.circleci/config.yml

```diff
@@ -1,41 +1,86 @@
+  dev-orb-version:
+    type: string
+    default: "dev:alpha"
+
+jobs:
+  pip-install-test:
+    executor: python/default
+    steps:
+        - run:
+              name: "Test pip"
+              command: echo "pytest >> requirements.txt"
+        - python/load-cache
+        - python/install-deps
+        - python/save-cache
+        - run:
+            name: "Test"
+            command: |
+              # pytest would be a dep in requirements.txt
+              pytest
 
 workflows:
-  main:
+  # This `lint-pack_validate_publish-dev` workflow will run on any commit.
+  lint_pack-validate_publish-dev:
+    unless: << pipeline.parameters.run-integration-tests >>
     jobs:
-      - test:
-          filters:
-            tags:
-              # Simplified SemVer regex
-              only: /^\d+\.\d+\.\d+$/
-      - publish:
-          filters:
-            branches:
-              ignore: /.*/
-            tags:
-              # Simplified SemVer regex
-              only: /^\d+\.\d+\.\d+$/
+      - orb-tools/lint
+      # pack your orb YAML files to a single orb.yml
+      # validate the orb.yml file to ensure it is well-formed
+      - orb-tools/pack:
+          requires:
+            - orb-tools/lint
+
+      - orb-tools/publish-dev:
+          orb-name: circleci/python
           context: orb-publishing
+          requires:
+            - orb-tools/pack
 
-jobs:
-  test:
-    executor: cli/default
-    steps:
-      - checkout
-      - run:
-          name: "Test & Validate Orb"
-          command: |
-            circleci config pack src > src/orb.yml
-            circleci orb validate src/orb.yml
-  publish:
-    executor: cli/default
-    steps:
-      - checkout
-      - run:
-          name: "Publish orb via the CircleCI CLI"
-          command: |
-            circleci config pack src > src/orb.yml
-            circleci orb publish src/orb.yml circleci/python@${CIRCLE_TAG}
+      # trigger an integration workflow to test the
+      # dev:${CIRCLE_SHA1:0:7} version of your orb
+      - orb-tools/trigger-integration-tests-workflow:
+          name: trigger-integration-dev
+          requires:
+            - orb-tools/publish-dev
+
+  # This `integration-tests_prod-release` workflow will only run
+  # when the run-integration-tests pipeline parameter is set to true.
+  # It is meant to be triggered by the "trigger-integration-tests-workflow"
+  # job, and run tests on <your orb>@dev:${CIRCLE_SHA1:0:7}.
+  integration-tests_prod-release:
+    when: << pipeline.parameters.run-integration-tests >>
+    jobs:
+      - pip-install-test
+      # publish a semver version of the orb. relies on
+      # the commit subject containing the text "[semver:patch|minor|major|skip]"
+      # as that will determine whether a patch, minor or major
+      # version will be published or if publishing should
+      # be skipped.
+      # e.g. [semver:patch] will cause a patch version to be published.
+      - orb-tools/dev-promote-prod-from-commit-subject:
```

Comment:
This should match the node orb nearly exactly, with different SSH fingerprints
https://github.com/CircleCI-Public/node-orb/blob/9079fe7ae040b459e80e0a3c096161d569f2d3a6/.circleci/config.yml#L193